ultralytics/yolo/engine/results.py

```diff
@@ -118,6 +123,53 @@ def __getattr__(self, attr):
                 orig_shape (tuple, optional): Original image size.
             """)
 
+    def visualize(self,
+                  img,
+                  show_conf=True,
+                  line_width=None,
+                  font_size=None,
+                  font='Arial.ttf',
+                  pil=False,
+                  example='abc'):
+        """
+        Plots the given result on an input RGB image. Accepts cv2(numpy) or PIL Image
+
+        Args:
+        img (): Image
+        show_conf (bool): Show confidence
+        line_width (Float): The line width of boxes. Automatically scaled to img size if not provided
+        font_size (Float): The font size of . Automatically scaled to img size if not provided
+        """
+        img = deepcopy(img)
+        if isinstance(img, Image.Image):  # handle PILLOW image
+            img = np.asarray(img)[:, :, ::-1]
+            img = np.ascontiguousarray(img)
+
+        annotator = Annotator(img, line_width, font_size, font, pil, example)
+        boxes = self.boxes
+        masks = self.masks
+        logits = self.probs
+        names = self.names
+
+        if boxes is not None:
```

Comment:
I think we should split these functionalities out to `visualize_box`, `visualize_mask`, visualize_prob`. Then users can visualize anything of these boxes, masks, probs as they want.

Reply:
Yeah that's the plat. Visualize() will visualize everything, visualize_box will only visualize box, and so on